metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +161,206 @@ Once we've chosen the implementation approach, let's consider the components dia
+
+##### SignInAnonymouslyUseCase
+> Explain the purpose and responsibility of the class.
+
+Since the public dashboard feature implies logging in as an anonymous user, we should create a use case needed to log in to the application using the anonymous user.
+
+##### AuthState
+> Explain the purpose and responsibility of the class.
+
+Since we should know the authentication state of the user, we should provide an enum that will contain all the following states:
+- loggedIn
+- loggedInAnonymously
+- loggedOut
+
+##### Auth module changes
+
+The list of changes this module requires to integrate the public dashboard feature:
+- add the `signInAnonymously()` method to the `UserRepository` abstract class;
+- add the implementation of the `signInAnonymously()` method to the `FirebaseUserRepository` class;
+- add the `isPublicDashboardFeatureEnabled` field of the boolean type to the `AuthNotifier`;
+- add the `isAuthInitialized` field of the boolean type to the `AuthNotifier`;  
+- add the `handlePublicDashboardFeatureConfigUpdates()` to the `AuthNotifier`, so we can handle the feature state updates;
+- create the [`UserProfileViewModel`](#UserProfileViewModel) class;
+- create the [`SignInAnonymouslyUseCase`](#SignInAnonymouslyUseCase) class, which calls the `signInAnonymously()` method of the `UserRepository`;
+- create the [`AuthState`](#AuthState) enum;
+- integrate the view model, the use case, and the enum described above into the `AuthNotifier`.
+
+Let's review the following class diagram, which describes the relationship between the classes in this module:
+
+![Auth Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_auth_class_diagram.puml)
+
+#### Navigation module
+
+The last thing, we should have the ability to manage the navigation of the anonymous user, that is, to know which pages they can visit and which they can't. The next subsection describes changes that we should implement to integrate the public dashboard feature.
+
+##### Navigation module changes
+
+The list of changes this module requires to integrate the public dashboard feature:
+- add the field of the `AuthState` type to the `NavigationNotifier`;
+- add the `handleAuthUpdates()` method to the `NavigationNotifier`, which should handle the updating of the `AuthState` in the `AuthNotifier`;
+- create a set of the pages' `RouteName`, which the anonymous user can visit (at the moment, it is only the `dashboard` route name);
```

Comment:
📱 
What do you think about adding the `bool` field into the `RouteConfiguration` class instead? Would it be a bit more clear and simple solution?